mesonbuild/cargo/manifest.py

```diff
@@ -0,0 +1,227 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright © 2022-2023 Intel Corporation
+
+"""Type definitions for cargo manifest files."""
+
+from __future__ import annotations
+import typing as T
+
+from typing_extensions import Literal, TypedDict, Required
+
+EDITION = Literal['2015', '2018', '2021']
+CRATE_TYPE = Literal['bin', 'lib', 'dylib', 'staticlib', 'cdylib', 'rlib', 'proc-macro']
+
+Package = TypedDict(
+    'Package',
+    {
+        'name': Required[str],
+        'version': Required[str],
+        'authors': T.List[str],
+        'edition': EDITION,
+        'rust-version': str,
+        'description': str,
+        'readme': str,
+        'license': str,
+        'license-file': str,
+        'keywords': T.List[str],
+        'categories': T.List[str],
+        'workspace': str,
+        'build': str,
+        'links': str,
+        'include': T.List[str],
+        'exclude': T.List[str],
+        'publish': bool,
+        'metadata': T.Dict[str, T.Dict[str, str]],
+        'default-run': str,
+        'autobins': bool,
+        'autoexamples': bool,
+        'autotests': bool,
+        'autobenches': bool,
+    },
+    total=False,
+)
+"""A description of the Package Dictionary."""
```

Comment:
That docstring seems misplaced.

Reply:
Or maybe not, when using the `Package = TypedDict()` syntax?

Reply:
this is a wierdness of vscode. If you put it here vscode will attach the docstring if you hover over it, like if you open cargo/interpreter.py and hover over it. If you put the comment above it, then it doesn't. Maybe that's too clever

Reply:
Ok, whatever works best... :)